actionpack/test/dispatch/debug_exceptions_test.rb

```diff
@@ -27,37 +27,37 @@ def call(env)
       env['action_dispatch.show_detailed_exceptions'] = @detailed
       req = ActionDispatch::Request.new(env)
       case req.path
-      when "/pass"
+      when %r{/pass}
         [404, { "X-Cascade" => "pass" }, self]
-      when "/not_found"
+      when %r{/not_found}
         raise AbstractController::ActionNotFound
-      when "/runtime_error"
+      when %r{/runtime_error}
         raise RuntimeError
-      when "/method_not_allowed"
+      when %r{/method_not_allowed}
         raise ActionController::MethodNotAllowed
-      when "/unknown_http_method"
+      when %r{/unknown_http_method}
         raise ActionController::UnknownHttpMethod
-      when "/not_implemented"
+      when %r{/not_implemented}
         raise ActionController::NotImplemented
-      when "/unprocessable_entity"
+      when %r{/unprocessable_entity}
         raise ActionController::InvalidAuthenticityToken
-      when "/not_found_original_exception"
+      when %r{/not_found_original_exception}
```

Comment:
Is there any reason why these strings have been converted into regex expressions? There are two patterns that start with the same line:

```ruby
when %r{/not_found}
  ...
when %r{/not_found_original_exception}
  ...
```

Because of the conflict, requests to `/not_found_original_exception` are being routed to the `/not_found` branch and they never hit what they are supposed to hit. As a result [this test](https://github.com/rails/rails/blob/1e25dfde032172bab2b126e60fac0302025defcc/actionpack/test/dispatch/debug_exceptions_test.rb#L316-L322) happened to be "passing". I noticed this when adding more assertions to it, but I think we should change them back to strings if there was not an explict reason to do so.